Support for TRIM Function

When converting COBOL code containing the TRIM function to Java, the following error occurs:
```sh
error: cannot find symbol
          CobolTerminal.display (0, 1, 1, CobolIntrinsic.funcTrim (0, 0, f_SAMPLE, 0));
                                                        ^
symbol:   method funcTrim(int,int,AbstractCobolField,int)
location: class CobolIntrinsic
1 error
```
It appears that this issue is due to the absence of the `funcTrim` method in the `CobolIntrinsic` class of the `libcobj`.

## Reproduction
Environment: Ubuntu 22.04 (WSL)

Code
```cobol
IDENTIFICATION DIVISION.
PROGRAM-ID. SAMPLE.

DATA DIVISION.
    WORKING-STORAGE SECTION.
        01 SAMPLE PIC X(6).  
PROCEDURE DIVISION.
    MAIN SECTION.
        MOVE " HELLO" TO SAMPLE.       
        DISPLAY SAMPLE.                
        DISPLAY FUNCTION TRIM(SAMPLE). 
```

Command
```sh
cobj -free sample.cbl
```

Thank you!

--- a/libcobj/app/src/main/java/jp/osscons/opensourcecobol/libcobj/common/CobolIntrinsic.java
+++ b/libcobj/app/src/main/java/jp/osscons/opensourcecobol/libcobj/common/CobolIntrinsic.java
@@ -2184,4 +2184,42 @@ public class CobolIntrinsic {
     currField.setInt(count);
     return currField;
   }
+
+  /** Equivalent to cob_intr_trim */
+  public static AbstractCobolField funcTrim(
+      int offset, int length, AbstractCobolField srcField, int direction) {
+    makeFieldEntry(srcField);
+    int i;
+    int srcFieldSize = srcField.getSize();
+    CobolDataStorage srcStorage = srcField.getDataStorage();
+    for (i = 0; i < srcFieldSize; ++i) {
+      if (srcStorage.getByte(i) != ' ') {
+        break;
+      }
+    }
+    if (i == srcFieldSize) {
+      currField.setSize(1);
+      currField.getDataStorage().setByte(0, (byte) ' ');
+      return currField;
+    }
+    int beginIndex = 0;
+    if (direction != 2) {
+      for (; srcStorage.getByte(beginIndex) == ' '; ++beginIndex)
+        ;
+    }
+    int endIndex = srcFieldSize - 1;
+    if (direction != 1) {
+      for (; srcStorage.getByte(endIndex) == ' '; --endIndex)
+        ;
+    }
+    CobolDataStorage currStorage = currField.getDataStorage();
+    currField.setSize(endIndex - beginIndex + 1);
+    for (i = 0; i <= endIndex - beginIndex; ++i) {
+      currStorage.setByte(i, srcStorage.getByte(beginIndex + i));
+    }
+    if (offset > 0) {
+      calcRefMod(currField, offset, length);
+    }
+    return currField;
+  }
 }

--- a/libcobj/app/src/main/java/jp/osscons/opensourcecobol/libcobj/common/CobolIntrinsic.java
+++ b/libcobj/app/src/main/java/jp/osscons/opensourcecobol/libcobj/common/CobolIntrinsic.java
@@ -2204,13 +2204,15 @@ public class CobolIntrinsic {
     }
     int beginIndex = 0;
     if (direction != 2) {
-      for (; srcStorage.getByte(beginIndex) == ' '; ++beginIndex)
-        ;
+      while (srcStorage.getByte(beginIndex) == ' ') {
+        ++beginIndex;
+      }
     }
     int endIndex = srcFieldSize - 1;
     if (direction != 1) {
-      for (; srcStorage.getByte(endIndex) == ' '; --endIndex)
-        ;
+      while (srcStorage.getByte(endIndex) == ' ') {
+        --endIndex;
+      }
     }
     CobolDataStorage currStorage = currField.getDataStorage();
     currField.setSize(endIndex - beginIndex + 1);

--- a/tests/run.src/functions.at
+++ b/tests/run.src/functions.at
@@ -1748,7 +1748,6 @@ AT_CHECK([java prog], [0],
 AT_CLEANUP
 
 AT_SETUP([FUNCTION TRIM])
-AT_CHECK([${SKIP_TEST}])
 
 AT_DATA([prog.cob], [
        IDENTIFICATION   DIVISION.
@@ -1773,7 +1772,6 @@ AT_CHECK([java prog], [0],
 AT_CLEANUP
 
 AT_SETUP([FUNCTION TRIM with reference modding])
-AT_CHECK([${SKIP_TEST}])
 
 AT_DATA([prog.cob], [
        IDENTIFICATION   DIVISION.

